src/main/java/me/gnat008/perworldinventory/data/FlatFile.java

```diff
@@ -41,21 +24,19 @@
 import java.io.IOException;
 import java.util.UUID;
 
-public class FileWriter implements DataWriter {
+public class FlatFile implements DataSource {
 
-    private final File FILE_PATH;
+    private final File DATA_FOLDER;
 
-    private final PerWorldInventory plugin;
-    private final PlayerSerializer playerSerializer;
-    private final PWIPlayerFactory pwiPlayerFactory;
+    private PerWorldInventory plugin;
+    private PlayerSerializer playerSerializer;
+    private PWIPlayerFactory pwiPlayerFactory;
```

Comment:
These fields are no longer set anywhere. They need to be either passed to the constructor as before or you need to make the data source provider have all of those classes so it can pass them along. If you want to do the first one (my recommendation), you can `@Inject Injector injector` into the data source provider and then call the appropriate class.

The advantage of a provider class is you can perform other stuff during the initialization process. In AuthMe, that's wrapping the data source with a cachedDataSource if needed, and checking the SQLite size to issue a warning in some cases. If you want to do something like that in the future, keep with the provider and either inject all required fields, or inject the injector. (You generally want to avoid injecting the injector, but for small contained classes like a provider, I'd argue that it's fine.)
If you don't plan on doing any additional work on the initialization of the data source, you could map each data source type to a class (add a field on the DataSourceType enum) and then just get the type from the settings, get the class and call the injector.